docker/linux/clang/Dockerfile

```diff
@@ -0,0 +1,159 @@
+ARG DOCKER_USERNAME
+ARG LIBSTDCPP_MAJOR_VERSION
+ARG DOCKER_TAG
+ARG DISTRO_VERSION
+ARG DISTRO
+ARG SUFFIX
+
+FROM ubuntu:${DISTRO_VERSION} as builder
+
+ARG CLANG_VERSION
+ARG LIBSTDCPP_VERSION
+
+LABEL maintainer="Ultimaker b.v. <software-cura@ultimaker.com>"
+
+RUN apt-get -qq update \
+    && apt-get -qq install -y --no-install-recommends --no-install-suggests -o=Dpkg::Use-Pty=0 \
+       apt-transport-https \
+       binutils \
+       wget \
+       curl \
+       ca-certificates \
+       git \
+       python3 \
+       python3-pip \
+    && pip3 install ninja cmake \
+    && ln -s /usr/bin/python3 /usr/bin/python \
+    && printf 'deb http://apt.llvm.org/xenial/   llvm-toolchain-xenial-10  main\n' > /etc/apt/sources.list.d/llvm.list \
+    && wget -q --no-check-certificate https://apt.llvm.org/llvm-snapshot.gpg.key \
+    && apt-key add llvm-snapshot.gpg.key \
+    && apt-get -qq update \
+    && apt-get -qq install -y --no-install-recommends --no-install-suggests -o=Dpkg::Use-Pty=0 clang-10 lld-10 libc++-10-dev \
+    && wget -q --no-check-certificate https://github.com/llvm/llvm-project/archive/llvmorg-${CLANG_VERSION}.tar.gz \
+    && tar zxf llvmorg-${CLANG_VERSION}.tar.gz \
+    && sed -E -i 's/OUTPUT_NAME\s+"unwind"/OUTPUT_NAME "llvm-unwind"/g' llvm-project-llvmorg-${CLANG_VERSION}/libunwind/src/CMakeLists.txt \
+    && sed -i 's/unwind/llvm-unwind/g' llvm-project-llvmorg-${CLANG_VERSION}/clang/lib/Driver/ToolChains/CommonArgs.cpp \
+    && cd llvm-project-llvmorg-${CLANG_VERSION} \
+    && mkdir build \
+    && cd build/ \
+    && cmake ../llvm \
```

Comment:
Why are we compiling LLVM here? Is `apt get clang` not sufficient? It would make this docker file a lot simpler, especially with having to mess with those temporary files below.